Source/EasyNetQ.Management.Client/ManagementClient.cs

```diff
@@ -588,35 +602,40 @@ public Task DeleteUserAsync(User user, CancellationToken cancellationToken = def
         {
             Ensure.ArgumentNotNull(vhost, nameof(vhost));
 
-            var result =
-                await GetAsync<AlivenessTestResult>($"aliveness-test/{SanitiseVhostName(vhost.Name)}",
-                    cancellationToken).ConfigureAwait(false);
+            var result = await GetAsync<AlivenessTestResult>($"aliveness-test/{SanitiseVhostName(vhost.Name)}",
+                cancellationToken).ConfigureAwait(false);
             return result.Status == "ok";
         }
 
-        private Task<T> GetAsync<T>(
+        private Task<TResult> GetAsync<TResult>(
             string path,
-            CancellationToken cancellationToken = default(CancellationToken),
-            params object[] queryObjects)
+            CancellationToken cancellationToken = default(CancellationToken))
         {
-            var request = CreateRequestForPath(path, HttpMethod.Get, queryObjects);
+            return GetAsync<TResult>(path, null, cancellationToken);
+        }
 
-            return httpClient.SendAsync(request, cancellationToken)
-                .ContinueWithOrThrow(_ => AnalyseResponse<T>(code => code == HttpStatusCode.OK, _.Result), cancellationToken)
-                    .ContinueWith(__ =>
-                    {
-                        request?.Dispose();
-                        return __;
-                    }, cancellationToken).Unwrap();
 
+        private Task<TResult> GetAsync<TResult>(
+            string path,
+            IReadOnlyDictionary<string, string> queryParameters,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
```

Comment:
As per my understanding we only have 3 types of *Criteria and we call GetAsync with 0, 1 or 2 of them (lengthsCriteria and ratesCriteria)
What if we overload ToString in the *Criteria classes and use a Linq Select/SelectMany plus a string.Join or simply a Linq Aggregate?

Reply:
Yep, there are only 3 criteria and we definitely can avoid reflection/code generation. 

PS I'm not sure about `ToString` methods, it might be better to return `IReadOnlyDictionary<string, string>` with `(name, value)` of properties.

Reply:
Both could be ok, but if we use dictionaries we need to have all criteria implementing an interface to be able to expose a method that does the conversion
For the merge probably it is possible to use Linq Union